### ISSUE TITLE
AsyncLightQueryAttribute doesn't work with .net core 6

### ISSUE THREAD

**Issue by mesut-aridoru** [GeorgDangl/LightQuery#40]
AsyncLightQueryAttribute doesn't work with .net core 6

        [HttpGet]
        [AsyncLightQuery(forcePagination: true)]
        [ProducesResponseType(typeof(PaginationResult<BuildingResponseModel>), 200)]
        public async Task<IActionResult> Get()
        {
            var results = await this._context
                  .Buildings
                  .ToListAsync();

            return this.Ok(results);
        }

**Comment by GeorgDangl** [comment]
Hey @mesut-aridoru, thank you for the issue!

With **LightQuery**, you have to return an `IQueryable` from your method. For you, it looks like it's a `List`.

The `AsyncLightQuery` attribute itself calls the async methods from EF Core, so all you need is to `return this.OK(this._context.Buildings)` and **LightQuery** will itself call async methods to produce the results. Like here: https://github.com/GeorgDangl/LightQuery/blob/dev/src/LightQuery.EntityFrameworkCore/AsyncLightQueryAttribute.cs#L62-L79

Basically, `[AsyncLightQuery]` is the same as `[LightQuery]`, but the async one internally uses the EF Core methods.

**Comment by GeorgDangl** [comment]
I've also just updated the README to make this a bit more clear: c335f6d3fa7c633589461d9ee895b5af1b5307ed

**Comment by mesut-aridoru** [comment]
thank you:)

**Comment by GeorgDangl** [comment]
Also, one more thing: You should probably add some ordering clause, and not just return directly the DbSet from the context. Otherwise, the order might not be consistent when you're paging.
